helix-core/src/comment.rs

```diff
@@ -9,6 +9,34 @@ use crate::{
 use helix_stdx::rope::RopeSliceExt;
 use std::borrow::Cow;
 
+pub const DEFAULT_COMMENT_TOKEN: &str = "//";
+
+/// Returns the longest matching comment token of the given line (if it exists).
+pub fn get_comment_token<'a, S: AsRef<str>>(
+    text: RopeSlice,
+    tokens: &'a [S],
+    line_num: usize,
+) -> Option<&'a str> {
+    let mut used_token: Option<&'a str> = None;
+
+    let line = text.line(line_num);
+    let start = line.first_non_whitespace_char()?;
+
+    let mut max_token_len = 0;
+    for token in tokens {
+        let token_str = token.as_ref();
+        let end = std::cmp::min(start + token_str.len(), line.len_chars());
+
+        let matches_token = line.slice(start..end).starts_with(token_str);
```

Comment:
This is mixing up the bytes/chars indices since `token_str.len()` is the number of bytes but we're treating it as the number of chars. It won't panic since we're capping it to the length in chars but if there were non-ASCII comment tokens there would be a correctness issue I believe.

We can use `slice(start..)` here anyways though and avoid computing `end` at all since we're using `starts_with`

Reply:
I believe this block can be simplified a bit using some iterator functions:

```rust
let line = text.line(line_num);
let start = line.first_non_whitespace_char()?;

tokens
    .iter()
    .map(AsRef::as_ref)
    .filter(|token| line.slice(start..).starts_with(token))
    .max_by_key(|token| token.len())
```